.github/workflows/ci.yml

```diff
@@ -100,9 +100,61 @@ jobs:
           exit 1
         fi
 
+  integration:
+    runs-on: ubuntu-latest
+    strategy:
+      # Prevent GitHub from cancelling all in-progress jobs when a matrix job fails.
+      fail-fast: false
+      matrix:
+        k8s: [v1.19, latest]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          override: true
+          toolchain: stable
+          profile: minimal
+      # Smart caching for Rust projects.
+      # Includes workaround for macos cache corruption.
+      # - https://github.com/rust-lang/cargo/issues/8603
+      # - https://github.com/actions/cache/issues/403
+      - uses: Swatinem/rust-cache@v1
+
+      - uses: nolar/setup-k3d-k3s@v1
+        with:
+          version: ${{matrix.k8s}}
```

Comment:
Testing against older Kubernetes version is a good start.

> Turns out we can define a minimum support and run simple tests against stable apis without changing k8s-openapi feature version at a start (because our tests don't rely on deprecated/alpha apis). 

Our tests don't use deprecated/alpha APIs currently, but don't we need to change `k8s-openapi`'s feature to prevent merging accidental changes that cause compile errors like the original one in #684?

Maybe we can just add a separate job that simply checks if `k8s-openapi/v1_XX` compiles.

Reply:
Good point, we do want to prevent that class of failures.

I might try to make an `e2e` test for this because it's hard to just compile `k8s-openapi/v1_XX` against `kube` without exposing tons of useless features on all our crates. In `e2e` we can instead have a single bin that can uses an older `k8s-openapi` without having to interact with our dev-deps.

Reply:
Ok, this is now done. We have an `e2e/boot.rs` test that does this, and some general improvements to ci has been done in general to make things nicer.